src/dbnode/client/session.go

```diff
@@ -1410,6 +1414,9 @@ func (s *session) writeAttemptWithRLock(
 			// towards quorum, current defaults, so this is ok consistency wise).
 			return
 		}
+		if hostShard.State() == shard.Initializing || hostShard.State() == shard.Leaving {
+			state.hostSucessMap[host.ID()] = false
```

Comment:
as above, the whole point of `ident.ID` is to avoid extra allocs. doing `host.ID()` is a lot more expensive, avoid that.